libs/community/tests/unit_tests/vectorstores/test_sqlserver.py

```diff
@@ -146,3 +146,110 @@ def test_that_a_document_entry_without_metadata_will_be_added_to_vectorstore(
     ]
     result = store.add_documents(docs)
     assert len(result) == len(docs)
+
+
+def test_sqlserver_delete_text_by_id_valid_ids_provided(
+    store: SQLServer_VectorStore,
+) -> None:
+    """Test that delete API deletes texts by id."""
+    texts = [
+        "Good review",
+        "new books",
+        "table",
+        "Sunglasses are a form of protective eyewear.",
+        "It's a new year.",
+    ]
+
+    metadatas = [
+        {"id": 100, "source": "book review", "length": 11},
+        {"id": 200, "source": "random texts", "length": 9},
+        {"id": 200, "source": "household list", "length": 5},
+        {"id": 600, "source": "newspaper page", "length": 44},
+        {"id": 300, "source": "random texts", "length": 16},
+    ]
+    store.add_texts(texts, metadatas)
+
+    result = store.delete(["100", "200", "600"])
+    # Should return true since valid ids are given
+    if result:
+        pass
```

Comment:
else how do you mark fail? Did you verify failure case?

Reply:
Failure will either
1. raise an error, and error case was tested when no ids are provided.
2. return false, when ids are not found.

Both are tested separately in other test cases.

Reply:
Updated case 1 to return false instead of an exception.